src/type/validate.ts

```diff
@@ -910,7 +910,7 @@ function createInputObjectDefaultValueCircularRefsValidator(
 
     // Check to see if there is cycle.
     const cycleIndex = fieldPathIndex[fieldStr];
-    if (cycleIndex !== undefined && cycleIndex > 0) {
+    if (cycleIndex !== undefined) {
```

Comment:
`cycleIndex` will never be <= 0 since it is only populated via `fieldPath.push()` which can never return a number less than `1`. I considered adding an invariant inside the if, but it didn't seem worthwhile.